src/main/java/nearsoft/academy/bigdata/recommendation/MovieRecommender.java

```diff
@@ -30,57 +30,63 @@
     private int reviews = 0;
     private long totalProd = 0;
     private long totalUsers = 0;
-    HashMap<String, Long> productsTotal = new HashMap<String, Long>();
-    HashMap<Long, String> productsTotalReverse = new HashMap<Long, String>();
-    HashMap<String, Long> usersTotal = new HashMap<String, Long>();
-    ArrayList<String> products = new ArrayList<String>();
-    ArrayList<String> users = new ArrayList<String>();
-    ArrayList<String> scores = new ArrayList<String>();
+    HashMap<String, Long> productsTotal = new HashMap<>();
+    HashMap<Long, String> productsTotalReverse = new HashMap<>();
+    HashMap<String, Long> usersTotal = new HashMap<>();
+    List<String> products = new ArrayList<>();
+    List<String> users = new ArrayList<>();
+    List<String> scores = new ArrayList<>();
     PrintWriter out = new PrintWriter(new FileWriter("movies.csv"));
 
     public MovieRecommender (String filename) throws IOException{
         this.filename = filename;
         getData();
```

Comment:
What do you think about having the `getData()` method call here? It means that each time you create a new instance of `MovieRecommender`, it will perform the logic for the recommender; is there some reason to have it here instead of calling it only when it's necessary to perform the operation? How this impact the design?

Reply:
Also. The name `getData` does not reflect the action performed inside the method